components/automate-gateway/gateway/datacollector.go

```diff
@@ -58,6 +58,14 @@ func (s *Server) dataCollectorHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	ticket, err := s.datacollectorLimiter.TakeTicket()
+	if err != nil {
+		s.logger.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusServiceUnavailable)
```

Comment:
Should we follow the mapping that grpc-gateway uses for the endpoints it provides? I believe that would mean 429 Too Many Requests,  https://github.com/googleapis/googleapis/blob/1b5a8d2bbb69c5a04db26bd546d2888e609c6bab/google/rpc/code.proto#L104-L108

Also, it would perhaps be more defensive to check if the returned error really is the one where there's no tickets left. At least from the interface, it could be anything that's going wrong.

Reply:
I dont think 429 is the correct one in this case as it isn't a limit applied to an individual client. The server is overloaded. My returning a 5xx, we tell the clients to come back. I'm not sure 429 would do that.

Ill fix up the errors to return the generic thing